File: android/app/src/main/kotlin/com/agent/aios/domain/repository/UpdateRepository.kt
package com.agent.aios.domain.repository

import com.agent.aios.domain.model.UpdateResult
import java.io.File

interface UpdateRepository {
    suspend fun checkForUpdate(): UpdateResult
    suspend fun downloadApk(url: String, fileName: String, onProgress: (Float) -> Unit): File?
    fun canInstallApk(): Boolean
    fun installApk(apkFile: File): Boolean
    fun requestInstallPermissionIntent(): android.content.Intent
}

File: lib/domain/repositories/update_repository.dart
import 'package:aios/domain/entities/update_info.dart';

abstract class UpdateRepository {
  Future<UpdateResult> checkForUpdate();
  Future<String?> downloadApk(
    String url,
    String fileName, {
    void Function(double progress)? onProgress,
  });
  Future<bool> installApk(String apkPath);
}
